CMakeLists.txt

```diff
@@ -387,49 +400,51 @@ endforeach()
 
 # TODO(JMM): Kind of nice to have?
 get_property(eos_headers GLOBAL PROPERTY EOS_HEADERS)
-get_property(eos_srcs GLOBAL PROPERTY EOS_SRCS)
+target_sources(singularity-eos_Interface PRIVATE ${eos_headers})
 message(VERBOSE "EOS Headers:\n\t${eos_headers}")
-message(VERBOSE "EOS Sources:\n\t${eos_srcs}")
 
-target_sources(singularity-eos PRIVATE ${eos_srcs} ${eos_headers})
+if(SINGULARITY_BUILD_CLOSURE)
```

Comment:
It might be best to have variable like `BUILD_LIBRARY` that (for now) automatically is set by `BUILD_CLOSURE` so that we can extend in the future if needed.